instrumentation/github.com/aws/aws-lambda-go/otellambda/xrayconfig/xrayconfig.go

```diff
@@ -89,14 +77,13 @@ func EventToCarrier() otellambda.Option {
 
 // Propagator returns an otellambda.Option to enable the xray.Propagator
 func Propagator() otellambda.Option {
-
 	return otellambda.WithPropagator(xray.Propagator{})
 }
 
 // AllRecommendedOptions returns a list of all otellambda.Option(s)
 // recommended for the otellambda package when using AWS XRay
-func AllRecommendedOptions() []otellambda.Option {
-	options, err := tracerProviderAndFlusher()
+func AllRecommendedOptions(tp *sdktrace.TracerProvider) []otellambda.Option {
```

Comment:
```suggestion
func RecommendedOptions(tp *sdktrace.TracerProvider) []otellambda.Option {
```

The `All` feels superfluous.  Can we drop it?